test/over_react/util/react_wrappers_test.dart

```diff
@@ -279,6 +353,34 @@ main() {
           expect(cloneDartProps, isNot(anyOf(contains('key'), contains('ref'))),
               reason: '"key" and "ref" should not be visible to the rendered cloned component');
         });
+
+        test('for a Dart component version 2', () {
+          ReactElement original;
+          ReactElement clone;
+
+          // The 'ref' property can only be used from within a render() method, so use RenderingContainerComponent
+          // to clone and render the test component.
+          var holder = RenderingContainerComponentFactory({
+            'renderer': () {
+              original = TestComponent2Factory(originalKeyRefPropsForComponent2,
+                  testChildren);
+              clone = cloneElement(original, overrideKeyRefProps);
+
+              return clone;
+            }
+          });
+          var renderedHolder = render(holder);
+
+          // Verify that "key" and "ref" are overridden according to React
+          expect(clone.key, overrideKeyRefProps['key']);
+          expect(cloneRefCalled, isTrue);
+
+          var renderedClone = react_test_utils.findRenderedComponentWithTypeV2(renderedHolder, TestComponent2Factory);
+
+          Map cloneDartProps = getDartComponent(renderedClone).props;
+          expect(cloneDartProps, isNot(anyOf(contains('key'), contains('ref'))),
+              reason: '"key" and "ref" should not be visible to the rendered cloned component');
```

Comment:
In Component2, we actually don't control whether those props make it to the component; React does instead. 

So, we can remove this expectation.